src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -0,0 +1,157 @@
+        mode: str,
+        encoding: str = Common.DEFAULT_ENCODING,
+        seek_position: int = 0,
+        data: Union[bytes, str] = "",
+    ) -> int:
+        """Write data in the file pointed to by the file-handle.
+
+        Args:
+            path (str): Path to the file to be opened
+            mode (str): Mode in whicg the file is to be opened. Usual options
+                        include r, rb, w and wb
+            encoding (str): Encoding type like utf-8 or utf-16
+            seek_position (int): Position to start writing from
+            data (Union[bytes, str]): Contents to be written
+
+        Returns:
+            int: Number of bytes that were successfully written to the file
+        """
+        try:
+            with self.fs.open(path=path, mode=mode, encoding=encoding) as file_handle:
+                return file_handle.write(data)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def seek_file(
+        self,
+        path: str,
+        location: int = 0,
+        position: FileSeekPosition = FileSeekPosition.START,
+    ) -> int:
+        """Place the file pointer to the mentioned location in the file
+        relative to the position.
+
+        Args:
+            path (str): path of the file
+            location (int): Nth byte position. To be understood in relation to
+            the arg "position"
+            position (FileSeekPosition): from start of file, current location
+            or end of file
+
+        Returns:
+            int: file pointer location after seeking to the mentioned position
+        """
+        try:
+            with self.fs.open(path=path, mode="rb") as file_handle:
+                return file_handle.seek(location, position)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def make_dir(self, path: str, create_parents: bool = True):
+        """Create a directory.
+
+        Args:
+            path (str): Path of the directory to be created
+            create_parents (bool): Specify if parent directories to be created
+            if any of the nested directory does not exist
+        """
+        try:
+            self.fs.mkdir(path, create_parents=create_parents)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def path_exists(self, path: str) -> bool:
+        """Checks if a file/directory path exists.
+
+        Args:
+            path (str): File/directory path
+
+        Returns:
+            bool: If the file/directory  exists or not
+        """
+        try:
+            return self.fs.exists(path)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def list(self, path: str) -> list[str]:
+        """List the directory path.
+
+        Args:
+            path (str): Directory path
+
+        Returns:
+            List[str]: List of files / directories under the path
+        """
+        try:
+            return self.fs.ls(path)
```

Comment:
@gaya3-zipstack What happens if a file path is passed here?

Reply:
@hari-kuriakose normal `ls` command will display the single files path in linux in this case. Similar is the behaviour with fsspec also.

<img width="581" alt="image" src="https://github.com/user-attachments/assets/1695f80a-f331-4de1-9bc6-f435b1d35dee">

Reply:
Added a test case and works as mentioned by @ritwik-g